zeroknowledgeproof/src/main/java/com/zeroknowledgeproof/numberProof/NumberProofProver.kt

```diff
@@ -0,0 +1,47 @@
+package com.zeroknowledgeproof.numberProof
+
+import java.math.BigInteger
+import java.math.BigInteger.ONE
+import java.math.BigInteger.ZERO
+import java.security.SecureRandom
+
+class NumberProofProver(val that: NumberProofTrustedParty, val N: BigInteger, val givenS: Int?= null) {
```

Comment:
Even though this pr is for the range proof mostly, we can drop the val for givenS.
we only use it to initialize s, so we don't have to store it as a field so: 

givenS: Int? =null instead of val givenS: Int? =null